src/tqec/compile/graph.py

```diff
@@ -374,6 +378,17 @@ def add_pipe(
         if block.is_temporal_pipe:
             self._check_block_spatial_shape(block)
             self._replace_temporal_borders(source, sink, block)
+            block_trimmed_temporal_borders = block.with_temporal_borders_replaced(
+                {
+                    TemporalBlockBorder.Z_NEGATIVE: None,
+                    TemporalBlockBorder.Z_POSITIVE: None,
+                }
+            )
+            if block_trimmed_temporal_borders:
+                u_pos = LayoutPosition3D.from_block_position(source)
```

Comment:
Now that temporal pipes might have a space-time existence, it might be nice to change the implementation of `LayoutPosition3D` to also be able to represent temporal pipes. 
Another solution would be to document the convention used above `self._temporal_pipes_at_hadamard_layer` to explicitly write that the temporal pipe should be placed just *after* (i.e., `z + 0.5`) the cube that is used as key.